### ISSUE TITLE
Mocks breaks in crystal 0.16.0

### ISSUE THREAD

**Issue by vjdhama** [waterlink/mocks.cr#20]
I was trying to run [ambience](https://github.com/vjdhama/ambience) specs against latest crystal release(0.16.0) locally, but it throws error.

```
$ crystal spec
Error in ./libs/mocks/mocks/registry.cr:140: can't use class variables in generic types

      @@_instances ||= reset!
      ^
```

I think it's because of these [changes](https://github.com/crystal-lang/crystal/blob/master/spec/compiler/type_inference/class_var_spec.cr#L146) in crystal. I'll take stab at fixing it and raise A PR if I can.

Cheers.

**Comment by marceloboeira** [comment]
@vjdhama thanks, that would be very helpful!

**Comment by vjdhama** [comment]
@waterlink I need help with this issue.

**Comment by waterlink** [comment]
@vjdhama Sure, let's fix that. Have you already advanced towards fixing? Can you share where you are code-wise now (like a link on github to a branch on your fork)?

**Comment by vjdhama** [comment]
@waterlink Sorry for late reply. I have pushed changes to my fork.

**Comment by vjdhama** [comment]
I still don't have a way to get all instances of Method(T) inside module `Instances` class method.

**Comment by strzibny** [comment]
This would be great to resolve as I am also stuck with this.

**Comment by waterlink** [comment]
@strzibny @vjdhama @marceloboeira I have just made a release `v0.9.2` and it should work with Crystal `0.18.4` now. Could you please try it out on your projects?

**Comment by marceloboeira** [comment]
@waterlink I'll check out :D

**Comment by strzibny** [comment]
@waterlink it doesn't for me :(

```
$ crystal spec ./spec/**
Error in ./libs/mocks/mocks/registry.cr:140: can't use class variables in generic types

      @@_instances ||= reset!
      ^

$ crystal -v
Crystal 0.18.7 [68783f1] (2016-07-03)

$ cat shard.lock 
version: 1.0
shards:
  kemal:
    github: sdogruyol/kemal
    version: 0.13.0

  kilt:
    github: jeromegn/kilt
    version: 0.3.3

  mocks:
    github: waterlink/mocks.cr
    version: 0.9.2

  radix:
    github: luislavena/radix
    version: 0.3.0

  redis:
    github: stefanwille/crystal-redis
    version: 1.6.5

  singleton:
    github: waterlink/singleton.cr
    version: 0.1.0
```

Here is my project if you want to try it yourself: https://github.com/strzibny/hipparchus

**Comment by waterlink** [comment]
Code that is seen in the error message is no longer in the library. I am assuming that you still have an old version. 

Can you show us your `shard.lock`?

`shards update` should help.

**Comment by strzibny** [comment]
@waterlink I showed the shard.lock in the comment, that's why I though I have correct version. But as you say the code was not updated. `crystal deps update`  fixed this, so I confirm this works :). Thank you!

**Comment by justaskz** [comment]
I am getting this on Crystal 0.18.4 and latest shard versions.

```
Error in ./libs/spec2-mocks/spec2-mocks.cr:34: can't declare variable of generic non-instantiated type Mocks::HaveReceivedExpectation(T)

    @unwrap : ::Mocks::HaveReceivedExpectation
```

**Comment by waterlink** [comment]
Please create the issue at spec2-mocks project
Justas Kazakauskas notifications@github.com schrieb am Do., 7. Juli 2016
um 3:37 PM:

**Comment by vjdhama** [comment]
@waterlink I still get an error, though that seems be due to `spec2-mock`.

```
Error in ./libs/spec2-mocks/spec2-mocks.cr:34: can't declare variable of generic non-instantiated type Mocks::HaveReceivedExpectation(T)

    @unwrap : ::Mocks::HaveReceivedExpectation
    ^~~~~~~
```

**Comment by vjdhama** [comment]
@waterlink Do we still have to use `spec2-mock` for `spec2` shard to play well with `mocks` shard?

**Comment by waterlink** [comment]
Yes. `spec2` should not know anything about `mocks`, as `mocks` should not know anything about `spec2`. `spec2-mocks` serves as a bridge between them and it knows about both of them, so that they can be changed independently (relatively).

**Comment by waterlink** [comment]
I have just released `v0.4.0` of `spec2-mocks` and it should work with newest `spec2` and `mocks`.

**Comment by vjdhama** [comment]
Kudos @waterlink. 👏
